packages/block-library/src/post-comment-author/edit.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+
+export default function Edit( { attributes, context } ) {
+	const { className } = attributes;
+	const { commentId } = context;
+
+	const displayName = useSelect( ( select ) => {
+		const { getEntityRecord } = select( 'core' );
+
+		const comment = getEntityRecord( 'root', 'comment', commentId );
+		if ( comment && ! comment.authorName ) {
+			const user = getEntityRecord( 'root', 'user', comment.author );
+
+			if ( ! user || ! user.name ) {
+				return __( 'Anonymous' );
+			}
+
+			return user.name;
```

Comment:
I think these could be shortened to

```suggestion
			return user?.name ?? __( 'Anonymous' );
```